pkg/virtctl/expose/expose.go

```diff
@@ -59,262 +61,234 @@ Possible types are (case insensitive, both single and plurant forms):
-			return fmt.Errorf("service creation failed for k8s < 1.20: %v", err)
-		}
+func convertServiceType(strServiceType string) (k8sv1.ServiceType, error) {
+	switch strings.ToLower(strServiceType) {
+	case strings.ToLower(string(k8sv1.ServiceTypeClusterIP)):
+		return k8sv1.ServiceTypeClusterIP, nil
+	case strings.ToLower(string(k8sv1.ServiceTypeNodePort)):
+		return k8sv1.ServiceTypeNodePort, nil
+	case strings.ToLower(string(k8sv1.ServiceTypeLoadBalancer)):
+		return k8sv1.ServiceTypeLoadBalancer, nil
+	case strings.ToLower(string(k8sv1.ServiceTypeExternalName)):
+		return "", fmt.Errorf("type: %s not supported", strServiceType)
+	default:
+		return "", fmt.Errorf("unknown service type: %s", strServiceType)
 	}
-	fmt.Printf("Service %s successfully exposed for %s %s\n", serviceName, vmType, vmName)
-	return nil
 }
 
-func convertIPFamily(strIPFamily string) ([]v1.IPFamily, error) {
+func convertIPFamily(strIPFamily string) ([]k8sv1.IPFamily, error) {
 	switch strings.ToLower(strIPFamily) {
 	case "":
-		return []v1.IPFamily{}, nil
-	case "ipv4":
-		return []v1.IPFamily{v1.IPv4Protocol}, nil
-	case "ipv6":
-		return []v1.IPFamily{v1.IPv6Protocol}, nil
-	case "ipv4,ipv6":
-		return []v1.IPFamily{v1.IPv4Protocol, v1.IPv6Protocol}, nil
-	case "ipv6,ipv4":
-		return []v1.IPFamily{v1.IPv6Protocol, v1.IPv4Protocol}, nil
+		return []k8sv1.IPFamily{}, nil
+	case strings.ToLower(string(k8sv1.IPv4Protocol)):
+		return []k8sv1.IPFamily{k8sv1.IPv4Protocol}, nil
+	case strings.ToLower(string(k8sv1.IPv6Protocol)):
+		return []k8sv1.IPFamily{k8sv1.IPv6Protocol}, nil
+	case strings.ToLower(string(k8sv1.IPv4Protocol) + "," + string(k8sv1.IPv6Protocol)):
+		return []k8sv1.IPFamily{k8sv1.IPv4Protocol, k8sv1.IPv6Protocol}, nil
+	case strings.ToLower(string(k8sv1.IPv6Protocol) + "," + string(k8sv1.IPv4Protocol)):
+		return []k8sv1.IPFamily{k8sv1.IPv6Protocol, k8sv1.IPv4Protocol}, nil
 	default:
 		return nil, fmt.Errorf("unknown IPFamily/s: %s", strIPFamily)
 	}
 }
 
-func convertIPFamilyPolicy(strIPFamilyPolicy string, ipFamilies []v1.IPFamily) (v1.IPFamilyPolicyType, error) {
+func convertIPFamilyPolicy(strIPFamilyPolicy string, ipFamilies []k8sv1.IPFamily) (k8sv1.IPFamilyPolicy, error) {
 	switch strings.ToLower(strIPFamilyPolicy) {
 	case "":
 		if len(ipFamilies) > 1 {
-			return v1.IPFamilyPolicyPreferDualStack, nil
+			return k8sv1.IPFamilyPolicyPreferDualStack, nil
 		}
 		return "", nil
-	case "singlestack":
-		return v1.IPFamilyPolicySingleStack, nil
-	case "preferdualstack":
-		return v1.IPFamilyPolicyPreferDualStack, nil
-	case "requiredualstack":
-		return v1.IPFamilyPolicyRequireDualStack, nil
+	case strings.ToLower(string(k8sv1.IPFamilyPolicySingleStack)):
+		return k8sv1.IPFamilyPolicySingleStack, nil
+	case strings.ToLower(string(k8sv1.IPFamilyPolicyPreferDualStack)):
+		return k8sv1.IPFamilyPolicyPreferDualStack, nil
+	case strings.ToLower(string(k8sv1.IPFamilyPolicyRequireDualStack)):
+		return k8sv1.IPFamilyPolicyRequireDualStack, nil
```

Comment:
nit: I would keep hardcoded optional strings (e.g: singlestack, ipv4) instead of generte them each time using ToLower.
Or initialize a map with all the names and qurey it where necessary, it could save from generating these strings every time.

Not a must.

Reply:
These funcs are all called once per command run so I expect there is not much to gain. In general, if these funcs could be called multiple times I'd agree.